### PR TITLE
Ofmcc 2036 pop up notification

### PR DESCRIPTION
**Pull request by jenbeckett**

### CODE REVIEW

File: frontend/src/views/HomeView.vue [hunk 1]
```diff
@@ -57,17 +62,92 @@
...
+    goToFunding() {
+      this.$router.push({ name: 'funding-overview' })
+    },
+    async loadFundingAgreements() {
+      this.isLoading = true
+
+      //I use for of here so I can break out of the loop when I hit 3 facilities.
+      //We show 3 max so the popup doesn't take over the screen, and we save on calls to Dynamics.
+      for (const fac of this.userInfo.facilities) {
+        const fa = await FundingAgreementService.getActiveFundingAgreementByFacilityId(fac.facilityId)
```
Comment (vietle-cgi) [inline frontend/src/views/HomeView.vue:135]: I think we only need to get the list of Signature Pending FA? In that case, we can create a new service function in FAservice to get FA having a specific status (e.g.: getActiveFundingAgreementByFacilityIdAndStatus) and include statusCode = Signature Pending in the request filter.

**Review by vietle-cgi — CHANGES_REQUESTED** [review]

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/views/HomeView.vue (hunk 1 above)
Comment (jenbeckett) [reply frontend/src/views/HomeView.vue:135]: done

File: frontend/src/views/HomeView.vue [hunk 2]
```diff
@@ -57,17 +62,92 @@
       </v-col>
     </v-row>
   </v-container>
+
+  <AppDialog v-model="isDisplayed" title="Signature Required!" persistent max-width="60%" @close="closeDialog">
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/HomeView.vue:66]: done

File: frontend/src/views/HomeView.vue [hunk 3]
```diff
@@ -57,17 +62,92 @@
...
 
 export default {
   name: 'HomeView',
-  components: { AppHeroImage, OrganizationHeader },
+  components: { AppHeroImage, AppButton, AppDialog, OrganizationHeader },
   mixins: [permissionsMixin],
+  data() {
+    return {
+      isLoading: false,
+      isDisplayed: false,
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/HomeView.vue:111]: done

File: frontend/src/views/HomeView.vue [hunk 4]
```diff
@@ -57,17 +62,92 @@
...
+      fundingAgreements: [],
+    }
+  },
+  computed: {
+    ...mapState(useAuthStore, ['userInfo']),
+  },
+  async created() {
+    this.MAX_FACILITIES_DISPLAYED = 3
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/HomeView.vue:119]: done

File: frontend/src/views/HomeView.vue [hunk 5]
```diff
@@ -57,17 +62,92 @@
...
 import permissionsMixin from '@/mixins/permissionsMixin.js'
 
 export default {
   name: 'HomeView',
-  components: { AppHeroImage, OrganizationHeader },
+  components: { AppHeroImage, AppButton, AppDialog, OrganizationHeader },
   mixins: [permissionsMixin],
+  data() {
+    return {
+      isLoading: false,
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/HomeView.vue:110]: done

File: frontend/src/views/HomeView.vue [hunk 6]
```diff
@@ -57,17 +62,92 @@
...
+      isDisplayed: false,
+      fundingAgreements: [],
+    }
+  },
+  computed: {
+    ...mapState(useAuthStore, ['userInfo']),
+  },
+  async created() {
+    this.MAX_FACILITIES_DISPLAYED = 3
+    await this.loadFundingAgreements()
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/HomeView.vue:120]: I agree! Probably wouldn't be worth even showing the popup unless you are the person who can sign it. I added a TODO so we can take care of this later.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/components/funding/SignFundingPopup.vue [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+.home-card {
+  border-top: 5px solid #003366 !important;
+  min-height: 225px;
+}
+
+.home-card-disabled {
+  border-top: 5px solid #909090 !important;
+  min-height: 225px;
+}
+</style>
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/components/funding/SignFundingPopup.vue:104]: done

File: frontend/src/views/HomeView.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/HomeView.vue]: ok I switched it back !

File: frontend/src/components/funding/SignFundingPopup.vue [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    await this.loadFundingAgreements()
+  },
+  methods: {
+    closeDialog() {
+      this.isDisplayed = false
+    },
+    goToFunding() {
+      this.$router.push({ name: 'funding-overview' })
+    },
+    async loadFundingAgreements() {
```
Comment (vietle-cgi) [inline frontend/src/components/funding/SignFundingPopup.vue:67]: should we add a try catch for this function?

**Review by vietle-cgi — CHANGES_REQUESTED** [review]

**Review by vietle-cgi — APPROVED** [review]

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**